src/struct/utils.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-restricted-syntax */
+type StructField = ByteType | ArrayType | StructType
+
+interface ByteType {
+  name: 'byte'
+  type: 'byte'
+  alias: string
+  byteLength: number
+}
+
+interface ArrayType {
+  name: string
+  type: 'array'
+  alias: string
+  item: StructField
+  itemCount: number
+  byteLength: number
+}
+
+interface StructType {
+  name: string
+  type: 'struct'
+  alias: string
+  fields: StructField[]
+  byteLength: number
+}
+
+const generateAlias = (field: StructField) => {
+  if (field.type === 'struct') {
+    const struct: StructType = field
+    for (const sub of struct.fields) {
+      sub.alias = `${sub.name ? sub.name : sub.type}-${field.alias}`
+      if (sub.type !== 'byte') {
+        generateAlias(sub)
+      }
+    }
+  } else if (field.type === 'array') {
+    const array: ArrayType = field
+    array.item.alias = `${array.item.name ? array.item.name : array.item.type}-${array.alias}`
+    if (array.item.type !== 'byte') {
+      generateAlias(array.item)
+    }
+  }
+}
+
+const normalizeAlias = (field: StructField) => {
+  field.alias = field.type === 'byte' ? field.type : field.name
+  generateAlias(field)
+}
+
+const aliasLengthPair: { alias: string; byteLength: number }[] = []
```

Comment:
The same problem as mentioned before. The `aliasLengthPair` will cache data pushed by `computeByteLength`, which means when the `computeByteLength` is called multiple times, the `aliasLengthPair` will mix them together.

And the mixture may have conflicts.